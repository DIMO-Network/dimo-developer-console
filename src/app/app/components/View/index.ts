import View from './View';

export { View };

export default View;
