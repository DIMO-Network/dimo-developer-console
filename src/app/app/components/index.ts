export * from './OnboardingCard';
export * from './Onboarding';
