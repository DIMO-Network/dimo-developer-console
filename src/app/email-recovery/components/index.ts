export * from './EmailRecoveryForm';
export * from './CheckEmail';