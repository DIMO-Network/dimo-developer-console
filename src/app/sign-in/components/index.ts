export * from './OtpInputForm';
export * from './PasskeyLogin';
export * from './SignInMethodForm';
