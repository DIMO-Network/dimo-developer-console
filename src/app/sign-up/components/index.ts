export * from './BuildForForm';
export * from './CompanyInfoForm';
export * from './SignUpWith';
export * from './UserInfoForm';
