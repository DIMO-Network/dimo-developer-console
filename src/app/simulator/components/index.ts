export { View } from './View';
export { Header } from './Header';
export { SimulatedVehiclesSection } from './SimulatedVehiclesSection';
export { CreateSimulatedVehicleModal } from './CreateSimulatedVehicleModal';
