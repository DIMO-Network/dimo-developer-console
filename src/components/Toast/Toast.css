.toast {
  @apply pointer-events-auto w-full max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5 z-30;

  &-content {
    @apply flex items-start p-4;

    &-content {
      @apply ml-3 w-0 flex-1 pt-0.5;
    }
  }

  &-icon-content {
    @apply flex-shrink-0;
  }

  &-title {
    @apply text-sm font-medium text-gray-900;
  }

  &-description {
    @apply mt-1 text-sm text-gray-500;
  }

  &-close-content {
    @apply ml-4 flex flex-shrink-0;

    .toast-close-btn {
      @apply inline-flex rounded-md bg-white text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2;
    }
  }
}
