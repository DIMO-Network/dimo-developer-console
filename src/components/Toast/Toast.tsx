'use client';

import { FC, Fragment, useState } from 'react';
import { Transition } from '@headlessui/react';
import {
  CheckCircleIcon,
  XMarkIcon as XMarkIcon24,
} from '@heroicons/react/24/outline';
import { XMarkIcon } from '@heroicons/react/20/solid';

import { INotification } from '@/hooks';

import './Toast.css';

export const Toast: FC<INotification> = ({ title, message, type }) => {
  const [show, setShow] = useState(true);

  return (
    <>
      <Transition
        show={show}
        as={Fragment}
        enter="transform ease-out duration-300 transition"
        enterFrom="translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2"
        enterTo="translate-y-0 opacity-100 sm:translate-x-0"
        leave="transition ease-in duration-100"
        leaveFrom="opacity-100"
        leaveTo="opacity-0"
      >
        <div className="toast">
          <div className="toast-content">
            <div className="toast-icon-content">
              {type === 'success' && (
                <CheckCircleIcon
                  className="h-6 w-6 text-green-400"
                  aria-hidden="true"
                />
              )}
              {type === 'error' && (
                <XMarkIcon24
                  className="h-6 w-6 text-red-400"
                  aria-hidden="true"
                />
              )}
            </div>
            <div className="toast-content-content">
              <p className="toast-title">{title}</p>
              <p className="toast-description">{message}</p>
            </div>
            <div className="toast-close-content">
              <button
                type="button"
                className="toast-close-btn"
                onClick={() => {
                  setShow(false);
                }}
              >
                <span className="sr-only">Close</span>
                <XMarkIcon className="h-5 w-5" aria-hidden="true" />
              </button>
            </div>
          </div>
        </div>
      </Transition>
    </>
  );
};

export default Toast;
