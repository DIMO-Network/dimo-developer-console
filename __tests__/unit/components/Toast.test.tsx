import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Toast } from '@/components/Toast';

describe('Toast', () => {
  it('renders a success notification', () => {
    const { container } = render(
      <Toast
        id={1234}
        message="Success information"
        type="success"
        title="Success"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders an error notification', () => {
    const { container } = render(
      <Toast
        id={1234}
        message="Error information"
        type="error"
        title="Error"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should close the notification', async () => {
    render(
      <Toast
        id={1234}
        message="Success information"
        type="success"
        title="Success"
      />,
    );

    const closeElm = await screen.findByRole('close-toast');
    fireEvent.click(closeElm);

    waitFor(() => {
      expect(closeElm).not.toBeInTheDocument();
    });
  });
});
