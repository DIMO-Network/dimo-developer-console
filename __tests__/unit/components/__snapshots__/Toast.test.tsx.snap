// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders a success notification 1`] = `
<div>
  <div
    class="toast"
  >
    <div
      class="toast-content"
    >
      <div
        class="toast-icon-content"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6 text-green-400"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 12.75 11.25 15 15 9.75M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="toast-content-content"
      >
        <p
          class="toast-title"
        >
          Success
        </p>
        <p
          class="toast-description"
        >
          Success information
        </p>
      </div>
      <div
        class="toast-close-content"
      >
        <button
          class="toast-close-btn"
          role="close-toast"
          type="button"
        >
          <span
            class="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.28 5.22a.75.75 0 0 0-1.06 1.06L8.94 10l-3.72 3.72a.75.75 0 1 0 1.06 1.06L10 11.06l3.72 3.72a.75.75 0 1 0 1.06-1.06L11.06 10l3.72-3.72a.75.75 0 0 0-1.06-1.06L10 8.94 6.28 5.22Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast renders an error notification 1`] = `
<div>
  <div
    class="toast"
  >
    <div
      class="toast-content"
    >
      <div
        class="toast-icon-content"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6 text-red-400"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18 18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="toast-content-content"
      >
        <p
          class="toast-title"
        >
          Error
        </p>
        <p
          class="toast-description"
        >
          Error information
        </p>
      </div>
      <div
        class="toast-close-content"
      >
        <button
          class="toast-close-btn"
          role="close-toast"
          type="button"
        >
          <span
            class="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            class="h-5 w-5"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.28 5.22a.75.75 0 0 0-1.06 1.06L8.94 10l-3.72 3.72a.75.75 0 1 0 1.06 1.06L10 11.06l3.72 3.72a.75.75 0 1 0 1.06-1.06L11.06 10l3.72-3.72a.75.75 0 0 0-1.06-1.06L10 8.94 6.28 5.22Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
